viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
+    watchlist = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for items in user_data["watchlist"]:
+        if title in user_data["watchlist"]:
+            user_data["watched"].append(user_data["watchlist"][0])
+            user_data["watchlist"].pop(0)
+        return user_data
+
+# def add_to_watched(user_data, movie):
+#     movies_watched = []
+#     updated_data = []
+
+#     for items in len(user_data)["movie"]:
+#         if movie not in user_data:
+#             user_data["watched"].append(updated_data[movies_watched])
+#         return updated_data
+
+        
+
+
+
+# make a dictionary with these keys above
+#title:title, genre:genre, rating:rating
+
+#-----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+#THIS TEST PASSED AND THEN STOPPED PASSING_ I"M NOT SURE WHY< BUT I DID NOT HAVE TIME TO FIX IT
+def get_watched_avg_rating(user_data):
+    ratings = []
+    ratings_sum = 0
+    if(len(user_data["watched"])) is 0:
+        ratings_average = 0.0
+    else:
+        for i in range(len(user_data["watched"])):
+            ratings.append((user_data["watched"][i]["rating"]))
+            i+=1
+        for rating in ratings:
+            ratings_sum += rating
+    ratings_average = ratings_sum / len(ratings)
+    return ratings_average
+
+def get_watched_avg_rating(user_data):
+    ratings = []
+    ratings_sum = 0
+    if(len(user_data["watched"])) is 0:
+        ratings_average = 0.0
+    return ratings_average
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+    genre_dictionary = {}
+
+    if (len(user_data["watched"])) is 0:
+        popular_genre = None
+    else: 
+        for i in range(len(user_data["watched"])):
+            genre_list.append((user_data["watched"][i] ["genre"]))
+            i+=1
+
+    for genre in genre_list:
+        count = genre_list.count(genre)
+        genre_dictionary[genre] = count
+
+    popular_genre = max(genre_dictionary, key=genre_dictionary.get)
+    return(popular_genre)
+
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+
+def get_unique_watched(user_data):
+
+
+    watched = user_data["watched"]
+    friends = user_data["friends"]
+    unique_movies = []
+    # movies_user_watched = watched(user_data)
+    # movies_friend_watched = friends(user_data)
+
+
+    watched.set(watched)
+    friends.set(friends)
+
+    for movie in watched:
+        if movie == (movie["title"], movie["genre"]):
+            unique_movies.append(movie)
+
+        return unique_movies
+
+def get_unique_friends_watched(user_data):
+    friends_unique_movies = []
+
+    friends_unique_movies
+
+
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):

```

Comment:
So right now, the tests are failing to run because of these empty function declarations. The tests not being able to run likely made it harder to debug your code.

This is reported in the test run in Learn in this section:

> E     File "/app/viewing_party/party.py", line 145
>
> E       def get_new_rec_by_genre(user_data):
> 
> E       ^
> 
> E   IndentationError: expected an indented block

In general when you want to stub functions out like this, make sure you include a stub block of only `pass` and you'll avoid that issue:

```python
def get_available_recs(user_data):
    pass
```

Reply:
Not that the reason the error actually shows up at the _next_ function, `get_new_rec_by_genre` is because that is where it was caught, since `get_available_recs` did not have the block that was expected.